core/src/main/java/juzu/impl/common/Tools.java

```diff
@@ -184,6 +184,9 @@ public boolean hasNext() {
             while (a != null && a.hasNext()) {
               try {
                 next = a.next();
+                if(next != null) {
```

Comment:
I guess the initial implementation just forgot this check in the while condition :
`while (next != null && a != null && a.hasNext())`
which avoids the return in the while loop.